plugins/analytics-module-ga/src/apis/implementations/AnalyticsApi/GoogleAnalytics.ts

```diff
@@ -151,4 +182,63 @@ export class GoogleAnalytics implements AnalyticsApi {
 
     return customDimensionsMetrics;
   }
+
+  /**
+   * Sets the GA userId, based on the `userEntityRef` set on the backstage
+   * identity loaded from a given Backstage Identity API instance. Because
+   * Google forbids sending any PII (including on the userId field), we hash
+   * the entire `userEntityRef` on behalf of integrators:
+   *
+   * - With value `User:default/name`, userId becomes `sha256(User:default/name)`
+   *
+   * If an integrator wishes to use an alternative hashing mechanism or an
+   * entirely different value, they may do so by passing a dummy Identity API
+   * implementation which returns a `userEntityRef` whose kind is the literal
+   * string `PrivateUser`, whose namespace is anything (it will be ignored) and
+   * whose name is the pre-hashed ID value.
+   *
+   * - With value `PrivateUser:default/a0n3b4n3`, userId becomes `a0n3b4n3`
+   * - With `PrivateUser:xyz/a0n3b4n3`, userId is `a0n3b4n3`
+   *
+   * Note: this feature requires that an integrator has set up a Google
+   * Analytics User ID view in the property used to track Backstage.
+   */
+  private async setUserFrom(identityApi: IdentityApi) {
+    const { userEntityRef } = await identityApi.getBackstageIdentity();
+
+    // Prevent PII from being passed to Google Analytics.
+    const userId = await this.getPrivateUserId(userEntityRef);
+
+    // Set the user ID.
+    ReactGA.set({ userId });
+
+    // Notify the deferred capture mechanism that it may proceed.
+    this.capture.setReady();
+  }
+
+  /**
+   * Returns a PII-free user ID for use in Google Analytics.
+   */
+  private getPrivateUserId(userEntityRef: string): Promise<string> {
+    const entity = parseEntityRef(userEntityRef);
+
+    // Mechanism allowing integrators to provide their own hashed values.
+    if (entity.kind === 'PrivateUser') {
```

Comment:
I'm feeling this is a little bit too magic tbh. How do you feel about these?

```ts
GoogleAnalytics.fromConfig(configApi, {
  identityApi,
  // default = 'sha-256'
  userIdTransform: 'none', // 'none' | 'sha-256' | (userEntityRef: string) => Promise<string>
})
```
or
```ts
GoogleAnalytics.fromConfig(configApi, {
  userId: identityApi.getBackstageIdentity().then(i => i.userEntityRef /* apply any transforms here */),
})
```

... with the possibility or replacing "userId" with any other sensible name, etc.

Reply:
I'm trying to stick with a convention of "analytics modules should accept `identityApi` and just use the `userEntityRef`" to .  Since GA is otherwise an exception on PII, I'm going with the `userIdTransform` approach.

Reply:
Updated here: 919a778